### ISSUE TITLE
find_by_sql add N charactor before and double 'single quote'

### ISSUE THREAD

**Issue by phanhongphucit** [rails-sqlserver/activerecord-sqlserver-adapter#439]
Hi,

I use gem Tidy_tds 0.7.0 and activerecord-sqlserver-adapter 3.2.13
When I call: 
EventLog.find_by_sql(["SELECT TOP 5 DATEADD(s, nDateTime, :zero_date), * FROM BioStar.dbo.tb_event_log", {:zero_date => '1970-01-01'}]) 
When execute it auto add N and double 'single quote' as follow error:
2016-01-22T08:40:04Z 10784 TID-nk1bk WARN: TinyTds::Error: Incorrect syntax near the keyword 'FROM'.: EXEC sp_executesql N' SELECT TOP 5 DATEADD(s, nDateTime, N''1970-01-01'') * FROM BioStar.dbo.tb_event_log '
2016-01-22T08:40:04Z 10784 TID-nk1bk WARN: /home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:416:in `each'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:416:in `handle_to_names_and_values_dblib'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:405:in `handle_to_names_and_values'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:377:in `_raw_select'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:371:in `block in raw_select'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:280:in `block in log'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activesupport-3.2.21/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:275:in `log'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:371:in `raw_select'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:354:in `do_exec_query'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:24:in `exec_query'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-sqlserver-adapter-3.2.13/lib/active_record/connection_adapters/sqlserver/database_statements.rb:297:in `select'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/database_statements.rb:18:in `select_all'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/query_cache.rb:63:in `select_all'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-3.2.21/lib/active_record/querying.rb:38:in `block in find_by_sql'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-3.2.21/lib/active_record/explain.rb:41:in `logging_query_plan'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/activerecord-3.2.21/lib/active_record/querying.rb:37:in `find_by_sql'
/home/administrator/workspace/sgh-testing-app/app/workers/fingerprint_importer.rb:32:in `perform'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/processor.rb:52:in `block (2 levels) in process'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/chain.rb:122:in `call'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/chain.rb:122:in `block in invoke'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/server/active_record.rb:6:in `call'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/chain.rb:124:in `block in invoke'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/server/retry_jobs.rb:62:in `call'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/chain.rb:124:in `block in invoke'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/server/logging.rb:11:in `block in call'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/logging.rb:22:in `with_context'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/server/logging.rb:7:in `call'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/chain.rb:124:in `block in invoke'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/chain.rb:127:in `call'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/middleware/chain.rb:127:in `invoke'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/processor.rb:51:in `block in process'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/processor.rb:94:in `stats'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/sidekiq-3.0.2/lib/sidekiq/processor.rb:50:in `process'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `public_send'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `dispatch'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/celluloid-0.16.0/lib/celluloid/calls.rb:122:in `dispatch'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/celluloid-0.16.0/lib/celluloid/cell.rb:60:in `block in invoke'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/celluloid-0.16.0/lib/celluloid/cell.rb:71:in `block in task'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/celluloid-0.16.0/lib/celluloid/actor.rb:357:in `block in task'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/celluloid-0.16.0/lib/celluloid/tasks.rb:57:in `block in initialize'
/home/administrator/.rvm/gems/ruby-2.0.0-p643@rails3.2/gems/celluloid-0.16.0/lib/celluloid/tasks/task_fiber.rb:15:in `block in create'

It takes me a day to trace but not found the problem yet, do you have any idea?

Thanks!

**Comment by theangryangel** [comment]
Your sql is missing a comma between the end of the date add function and the *

**Comment by phanhongphucit** [comment]
Thansk theangryangel and sorry, it's just my missing when posting. The error still raise when I add the comma.

**Comment by metaskills** [comment]
Then please update your description with the proper error and full stack trace. Also, please use GitHub's triple backticks to help format too.

**Comment by metaskills** [comment]
I noticed this was for v3.2.x of the adapter. I am open to bug fixes on this, but I am certain we are passing tests on our 3-2-stable branch. This means that any issues reported are likely either a) personal in your own stack or b) edge cases, normally beyond the scope of the adapter. In the even it is the later, I am open to fixes. I just don't have time to discuss or debug this issue on users' behalf. Closing this issue.
